### PR TITLE
fix: Improve Jupyter performance

### PR DESCRIPTION
**Pull request by DanielSchiavini**
### What I did
Improved the jupyter plugin performance considerably
Fixed the empty elements while calling javascript

### How I did it
- Minified JS with command `npx uglify-js --mangle -o jupyter.min.js -- jupyter.js`
- Instead of injecting the whole ethersJS library, just inline the wallet calls
- Only inject the JS triggers once for JupyterLab - in colab this does not work

### How to verify it
- Jupyter plugin should keep on working but much better

### Description for the changelog
- Improved and simplified JupyterLab and Colab plugin

### Cute Animal Picture
image

### CODE REVIEW

**Comment by charles-cooper** [comment]
i'm not convinced we want the complexity of introducing minification into the build process. fwiw after compression the minified version saves 900 bytes:
```
(boa) ~/titanoboa $ gzip -v < boa/integrations/jupyter/jupyter.js | wc
 61.7%
      6      47    1822
(boa) ~/titanoboa $ gzip -v < boa/integrations/jupyter/jupyter.min.js | wc
 47.0%
      3      19     949
```

File: boa/integrations/jupyter/browser.py [hunk 1]
```diff
@@ -49,12 +58,10 @@ def __init__(self, address=None):
...
         :param address: The account address. If not provided, it will be requested from the browser.
         """
-        if address is not None:
-            self.address = address
-        else:
-            self.address = _javascript_call(
-                "loadSigner", timeout_message=ADDRESS_TIMEOUT_MESSAGE
-            )
+        address = _javascript_call(
+            "loadSigner", address, timeout_message=ADDRESS_TIMEOUT_MESSAGE
```
[the remark this answers is not in the document]
Comment (DanielSchiavini) [reply boa/integrations/jupyter/browser.py:62]: yeah this works the same: https://github.com/vyperlang/titanoboa/blob/215fec9bca96113fad80654a42caf9f40a76d782/boa/integrations/jupyter/jupyter.js#L44

**Review by charles-cooper — APPROVED** [review]

**Comment by charles-cooper** [comment]
code review note: discussed refactoring the jupyter.js file into a full-on extension, but it seems to introduce more complexity than it is worth.

File: boa/integrations/jupyter/jupyter.js [hunk 2]
```diff
@@ -91,7 +95,8 @@
         }
 
         const body = stringify(await parsePromise(func(...args)));
-        // console.log(`Boa: ${func.name}(${args.map(a => JSON.stringify(a)).join(',')}) = ${body};`);
+        const duration = Date.now() - start;
```
Comment (charles-cooper) [inline boa/integrations/jupyter/jupyter.js:98]: dead variable

File: boa/integrations/jupyter/jupyter.js [hunk 3]
```diff
@@ -83,6 +86,7 @@
 
     /** Call the backend when the given function is called, handling errors */
     const handleCallback = func => async (token, ...args) => {
+        const start = Date.now();
```
Comment (charles-cooper) [inline boa/integrations/jupyter/jupyter.js:89]: also dead!

**Review by charles-cooper — APPROVED** [review]
lgtm after removing the dead variables

**Comment by charles-cooper** [comment]
just tested this -- works amazing! and it's deployed to try.vyperlang.org now :)

**Pull request merged**